confignode/src/main/java/org/apache/iotdb/confignode/persistence/NodeInfo.java

```diff
@@ -180,6 +179,7 @@ public TSStatus registerDataNode(RegisterDataNodePlan registerDataNodePlan) {
           nextNodeId.set(info.getLocation().getDataNodeId());
         }
       }
+      registeredDataNodes.put(info.getLocation().getDataNodeId(), info);
```

Comment:
Why we lacked this line before~

Reply:
Because we split the startup process of DataNode into double RPC requests in the former design, and this line was contained in the second RPC request.